src/extension.ts

```diff
@@ -100,3 +107,49 @@ class GdbAdapterDescriptorFactory implements vscode.DebugAdapterDescriptorFactor
         return new vscode.DebugAdapterInlineImplementation(this.debugSession);
     }
 }
+
+class gnuCobolEvalExpressionFactory implements EvaluatableExpressionProvider {
+
+	provideEvaluatableExpression(document: TextDocument, position: Position): ProviderResult<EvaluatableExpression> {
+		if (this.isCommentLine(document, position)) {
+			return undefined;
+		}
+		const selectionRange = this.getSelectionRangeInEditor();
+		if (selectionRange) {
+			return new EvaluatableExpression(selectionRange);
+		}
+		const wordRange = document.getWordRangeAtPosition(position)
+		return wordRange ? new EvaluatableExpression(wordRange) : undefined;
+	}
+
+	/**
+	 * Checks if the current line represents a comment line
+	 *
+	 * @param document document which is being evaluated
+	 * @param position position of the current line
+	 */
+	private isCommentLine(document: TextDocument, position: Position): boolean {
+		const start = new Position(position.line, 0);
+		const end = new Position(position.line, MAX_COLUMN_INDEX);
+		const range = new Range(start, end);
+		const lineText = document.getText(range);
+		return lineText.trimLeft().startsWith(COBOL_COMMENT_SYMBOL);
```

Comment:
it likely is reasonable to also explicit check column 7 (if the line has that) for a plain asterisk - most code is still in fixed-form reference-format

if it is possible (the `document` providing that) then it would be best to get the information from the editor (language provider), but I haven't checked if that is possible

Reply:
Adjusted. I also did the handling for free-form comments.